ecmascript/parser/tests/typescript/issue-896/input.ts.json

```diff
@@ -0,0 +1,29 @@
+{
+  "type": "Module",
+  "span": {
+    "start": 14,
+    "end": 45,
+    "ctxt": 0
+  },
+  "body": [
+    {
+      "type": "ExportAllDeclaration",
+      "span": {
+        "start": 14,
+        "end": 45,
+        "ctxt": 0
+      },
+      "source": {
+        "type": "StringLiteral",
+        "span": {
+          "start": 33,
+          "end": 45,
+          "ctxt": 0
+        },
+        "value": "../typings",
+        "hasEscape": false
+      }
+    }
+  ],
```

Comment:
Looks like it's the same AST node as for `export * from ".."` - so I guess there will be no way of telling if this is type only import/export. IMO it'd be very useful to have that information, in Deno we got `--no-check` option which emits type stripped code and elides type-only imports/exports. Currently Deno uses TSC to do that stripping but I intended to rewrite that functionality in SWC (https://github.com/denoland/deno/issues/6864), but I think it might not be possible without knowing that this statement can be safely elided.

Reply:
Unless I'm out of my way and `typescript::strip()` handles that automatically?

Reply:
I got it.

@bartlomieju 
How do you think about emitting error when user input is 
```ts
export type * from '../typings' 
```
It's invalid syntax, but error can be ignored using `@ts-ignore`

Reply:
To be honest I don't know, CCing @kitsonk for his expertise

Reply:
On the other hand, probably the least surprising behavior would be to duplicate what TSC does

Reply:
I'll create another pr for ast changes

Reply:
Babel and typescript-eslint set `exportKind` to `"type"` instead of `"value"`: https://astexplorer.net/#/gist/408ad6e07e866fd7743725e9073bdf5b/74e0f25919bf3f997a3d83dee2a3635fc701156a.  TypeScript `transpileOnly()` will always strip `export type` and `import type` statements.  The main reason the syntax was introduced was to make type stripping possible (as `export ... from ...` cannot be statically determined to be a type or not).

Erroring I suspect would be a challenge, since it is valid TypeScript and we wouldn't want to `@ts-ignore` it.

Reply:
tsc emits `TS1383: Only named exports may use 'export type'` for

```ts
export type * from '../typings'
```